### DISCUSSION TITLE
Drag and drop tasks within a group / Allow detaching tasks from their app group

### DISCUSSION THREAD

**Discussion by mattivl** [lawand-dot-io/taskbar#114] [Feature requests]
At the moment new windows (task) of an app open at the end of the taskbar, in case the "Group tasks by app (chronologically)" is turned off. I would like an option that they automatically open at the right end of the task group.
On the other hand side, when the "Group tasks by app (chronologically)" is turned on, it is not possible to drag an drop single windows/tasks of this group anymore.

My ideal scenario is thus:

1. Pin apps to certain positions in the taskbar.
  a. Means they are always visible, even when closed which makes it easy to open them.
  b. It also means that I always know more or less where to find these apps and its windows.
1. New windows of an app automatically open on the right end of an app group.
1. I can manually reorder windows inside an app group.
1. I can also move a window to a completely different spot in the taskbar, removing it from the app group. While new windows still open inside the app group.

**Comment by lawand-dot-io** [comment]
Hello. Thanks for the request. Here are some comments:

> 1. Pin apps to certain positions in the taskbar.
a. Means they are always visible, even when closed which makes it easy to open them.
b. It also means that I always know more or less where to find these apps and its windows.

a. this feature already exists, by right clicking a window and selecting the "Keep in Taskbar" option. Isn't that what you are looking for?
b. the feature you are referring to here is keeping apps in the same position whether or not they are open. This is mentioned in another feature request (https://github.com/lawand-dot-io/taskbar/discussions/76) and I will probably add it in a future update.

> 2. New windows of an app automatically open on the right end of an app group.
> 3. I can manually reorder windows inside an app group.

I had this in mind at some point. I should allow drag and drop within an app group.

> 4. I can also move a window to a completely different spot in the taskbar, removing it from the app group. While new windows still open inside the app group.

This might be too complicated. If you detach a window from an app group and move it into a new group, in this case we would have multiple app groups, so what do I do when a new window from the same app is opened? Which of the two group would I add this window to?

**Reply by mattivl** [reply]
Hello,

> a. this feature already exists, by right clicking a window and selecting the "Keep in Taskbar" option. Isn't that what you are looking for?
b. the feature you are referring to here is keeping apps in the same position whether or not they are open. This is mentioned in another feature request (https://github.com/lawand-dot-io/taskbar/discussions/76) and I will probably add it in a future update.


Yes, sorry for the confusing comment. I know that it already works. I just wanted to outline my whole workflow to clarify.

> I had this in mind at some point. I should allow drag and drop within an app group.

That would be great!

> This might be too complicated. If you detach a window from an app group and move it into a new group, in this case we would have multiple app groups, so what do I do when a new window from the same app is opened? Which of the two group would I add this window to?

I would rather say that one window is roaming freely and is not automatically getting part of another app group, when detached. New windows are still opened in the original (and only) app group.

**Reply by lawand-dot-io** [reply]
@mattivl Thanks for the info. I updated this request title to be "Drag and drop tasks within a group / Allow detaching tasks from their app group" which I believe includes all the things you are asking for. I will keep this request open and consider implementing it in the future, and as always the requests that are most upvoted are more likely to be implemented. Cheers
